lib/xml/reader.dart

```diff
@@ -141,3 +144,147 @@ class XmlReader {
     return context.success(null, context.position + 1);
   }
 }
+
+/// A push based XmlReader interface, intended to be similar to .NET's XmlReader.
+class XmlTextReader {
+  /// Creates a new reader for `input`.
+  /// 
+  /// Setting `ignoreWhitespace` to false will cause text nodes 
+  XmlTextReader(String input, {this.ignoreWhitespace = true}) {
+    _result = Success(input, 0, null);
+    _depth = 0;
+    _eof = false;
+  }
+
+  /// If true, will ignore `XmlNodeType.TEXT` and when it is composed of whitespace.
+  bool ignoreWhitespace;
```

Comment:
final? Where is this used?

Reply:
Right now, a consumer could change this during parsing but I'm not sure why they would.  I'm fine with making it final.  It's used when you have whitespace you want to ignore between nodes, e.g. `<xml>      <element />     </xml>`.  If this property is set to false, you'd get `XmlNodeType.ELEMENT` followed by `XmlNodeType.TEXT` (value = `      `), followed by `XmlNodeType.ELEMENT` (isEmptyElement = `true`), etc.

Most consumers probably don't want that whitespace, and the spec wouldn't consider it significant anyway.